src/openrct2/GameStateSnapshots.cpp

```diff
@@ -263,58 +242,102 @@ struct GameStateSnapshots final : public IGameStateSnapshots
-        COMPARE_FIELD(Peep, VoucherType);
-        COMPARE_FIELD(Peep, VoucherRideId);
-        COMPARE_FIELD(Peep, SurroundingsThoughtTimeout);
-        COMPARE_FIELD(Peep, Angriness);
-        COMPARE_FIELD(Peep, TimeLost);
-        COMPARE_FIELD(Peep, DaysInQueue);
-        COMPARE_FIELD(Peep, BalloonColour);
-        COMPARE_FIELD(Peep, UmbrellaColour);
-        COMPARE_FIELD(Peep, HatColour);
-        COMPARE_FIELD(Peep, FavouriteRide);
-        COMPARE_FIELD(Peep, FavouriteRideRating);
+    }
+
+    void CompareSpriteDataStaff(const Staff& spriteBase, const Staff& spriteCmp, GameStateSpriteChange_t& changeData) const
+    {
+        CompareSpriteDataPeep(spriteBase, spriteCmp, changeData);
+
+        COMPARE_FIELD(Staff, AssignedStaffType);
+        COMPARE_FIELD(Staff, MechanicTimeSinceCall);
+        COMPARE_FIELD(Staff, HireDate);
+        COMPARE_FIELD(Staff, StaffId);
+        COMPARE_FIELD(Staff, StaffOrders);
+        COMPARE_FIELD(Staff, StaffMowingTimeout);
+        COMPARE_FIELD(Staff, StaffRidesFixed);
+        COMPARE_FIELD(Staff, StaffRidesInspected);
+        COMPARE_FIELD(Staff, StaffLitterSwept);
+        COMPARE_FIELD(Staff, StaffBinsEmptied);
+    }
+
+    void CompareSpriteDataGuest(const Guest& spriteBase, const Guest& spriteCmp, GameStateSpriteChange_t& changeData) const
+    {
+        CompareSpriteDataPeep(spriteBase, spriteCmp, changeData);
+
+        COMPARE_FIELD(Guest, OutsideOfPark);
+        COMPARE_FIELD(Guest, GuestNumRides);
+        COMPARE_FIELD(Guest, Happiness);
+        COMPARE_FIELD(Guest, HappinessTarget);
+        COMPARE_FIELD(Guest, Nausea);
+        COMPARE_FIELD(Guest, NauseaTarget);
+        COMPARE_FIELD(Guest, Hunger);
+        COMPARE_FIELD(Guest, Thirst);
+        COMPARE_FIELD(Guest, Toilet);
+        COMPARE_FIELD(Guest, TimeToConsume);
+        COMPARE_FIELD(Guest, Intensity);
+        COMPARE_FIELD(Guest, NauseaTolerance);
+        COMPARE_FIELD(Guest, PaidOnDrink);
+        for (int i = 0; i < 16; i++)
+        {
+            COMPARE_FIELD(Guest, RideTypesBeenOn[i]);
+        }
+        COMPARE_FIELD(Guest, ItemFlags);
+        COMPARE_FIELD(Guest, Photo2RideRef);
+        COMPARE_FIELD(Guest, Photo3RideRef);
+        COMPARE_FIELD(Guest, Photo4RideRef);
+        COMPARE_FIELD(Guest, GuestNextInQueue);
+        COMPARE_FIELD(Guest, TimeInQueue);
+        for (int i = 0; i < 32; i++)
+        {
+            COMPARE_FIELD(Guest, RidesBeenOn[i]);
+        }
+
+        COMPARE_FIELD(Guest, CashInPocket);
+        COMPARE_FIELD(Guest, CashSpent);
+        COMPARE_FIELD(Guest, ParkEntryTime);
+        COMPARE_FIELD(Guest, RejoinQueueTimeout);
+        COMPARE_FIELD(Guest, PreviousRide);
+        COMPARE_FIELD(Guest, PreviousRideTimeOut);
+        for (int i = 0; i < PEEP_MAX_THOUGHTS; i++)
+        {
+            COMPARE_FIELD(Guest, Thoughts[i]);
+        }
+        COMPARE_FIELD(Guest, GuestHeadingToRideId);
+        COMPARE_FIELD(Guest, GuestIsLostCountdown);
```

Comment:
This didn't use to be on the comparisons :O. I'm assuming it was an oversight from the past.

Reply:
It was one of the few union fields that had the staff field first so it was there just under the staff name